math/mode_test.go

```diff
@@ -0,0 +1,114 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Test for Finding Mode Value In an Array
+
+package math
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestModeInteger(t *testing.T) {
+
+	testCases := []struct {
+		name    string
+		numbers []int
+		mode    int
+		err     error
+	}{
+		// Test Cases
+		// For INTEGER values
```

Comment:
This is a good comment. Can be in a single line though.

```suggestion
		// Test cases for `int` values
```